packages/sanity/src/structure/useStructureToolSetting.ts

```diff
@@ -9,36 +10,29 @@ export function useStructureToolSetting<ValueType>(
   key: string,
   defaultValue?: ValueType,
 ): [ValueType | undefined, (_value: ValueType) => void] {
-  const settingsStore = useSettingsStore()
+  const KVStore = useKVStore()
   const [value, setValue] = useState<ValueType | undefined>(defaultValue)
 
-  const structureToolSettings = useMemo(
-    () => settingsStore.forNamespace('structure-tool'),
-    [settingsStore],
-  )
+  const KVStoreKey = namespace ? `structure-tool::${namespace}::${key}` : `structure-tool::${key}`
 
   const settings = useMemo(() => {
-    if (namespace) {
-      return structureToolSettings.forNamespace(namespace).forKey(key)
-    }
-
-    return structureToolSettings.forKey(key)
-  }, [structureToolSettings, namespace, key])
+    return KVStore.getKey(KVStoreKey)
+  }, [KVStore, KVStoreKey])
 
   useEffect(() => {
-    const sub = settings.listen(defaultValue).subscribe({
+    const sub = settings.pipe(startWith(defaultValue)).subscribe({
       next: setValue as any,
     })
 
     return () => sub?.unsubscribe()
-  }, [defaultValue, key, namespace, settings])
+  }, [defaultValue, KVStoreKey, settings])
 
   const set = useCallback(
     (newValue: ValueType) => {
       setValue(newValue)
-      settings.set(newValue as any)
+      KVStore.setKey(KVStoreKey, newValue as string)
```

Comment:
Per my above comment, this is being cast a string since that's technically what is always getting stored (for now). Should we export the KVStoreValue type so it can be used by all consumers?